source/extensions/filters/http/connect_stats/connect_stats_filter.cc

```diff
@@ -0,0 +1,278 @@
+#include "source/extensions/filters/http/connect_stats/connect_stats_filter.h"
+
+#include <memory>
+
+#include "envoy/extensions/filters/http/connect_stats/v3/config.pb.h"
+#include "envoy/extensions/filters/http/connect_stats/v3/config.pb.validate.h"
+#include "envoy/grpc/context.h"
+#include "envoy/registry/registry.h"
+
+#include "source/common/grpc/common.h"
+#include "source/common/grpc/context_impl.h"
+#include "source/common/runtime/runtime_impl.h"
+#include "source/common/stats/symbol_table.h"
+#include "source/common/stream_info/utility.h"
+#include "source/extensions/filters/http/common/pass_through_filter.h"
+#include "source/extensions/filters/http/connect_stats/response_frame_counter.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace ConnectStats {
+
+namespace {
+
+// TODO(jchadwick-buf): This is a copy of GrpcServiceMethodToRequestNamesMap from
+// grpc_stats_filter.cc. Either this extension should be somehow merged with gRPC Stats, or at
+// least some code should probably be shared.
+class ConnectServiceMethodToRequestNamesMap {
```

Comment:
The duplication here may hint that it is better to simply merge this filter with `grpc_stats`. I'm open to that perspective even if the bridge filter remains independent. Of course, if there's another good way to reduce this duplication, I'm all-ears.

Reply:
I would vote to merge it, yes. I think there is so much overlap in config, etc. that it doesn't make sense to have a new filter for this part.

Reply:
Merged. Thanks.